### DISCUSSION TITLE
Gremlin vertex with list properties

### DISCUSSION THREAD

**Discussion by michaelawarren** [ArcadeData/arcadedb#212] [Q&A] [answered]
I'm trying to create a property with a list of values. It looks like there is support for this [DataTypes](https://docs.arcadedb.com/#DataTypes). This worked in Orient DB which ArcadeData says it is a fork of.  Is there a way to make this work with gremlin queries.

```groovy
g.addV().property(list, 'myList', ['a', 'b'])
```

**Comment by lvca** [comment, marked as answer]
Unbale to reproduce it. Tested this and works:

```java
@Test
  public void testGremlinLists() throws ExecutionException, InterruptedException {
    final ArcadeGraph graph = ArcadeGraph.open("./target/testlist");
    try {
      final ResultSet result = graph.gremlin("g.addV('Person').property( 'list', ['a', 'b'] )").execute();

      Assertions.assertTrue(result.hasNext());
      final Result v = result.next();
      Assertions.assertTrue(v.isVertex());
      final List list = (List) v.getVertex().get().get("list");
      Assertions.assertEquals(2, list.size());
      Assertions.assertTrue(list.contains("a"));
      Assertions.assertTrue(list.contains("b"));

    } finally {
      graph.drop();
    }
  }
```

**Reply by michaelawarren** [reply]
Ah i see what i was doing wrong. In orient db i was specifying the cardinality as the first value in properties(Cardnality.list, propertyname, propertyValue) (https://tinkerpop.apache.org/docs/current/reference/#addproperty-step). In arcade db this isn't necessary. Thank you for the quick responses.

**Reply by lvca** [reply]
Glad it worked out.
